src/WebJobs.Extensions.SendGrid/Bindings/SendGridMessageAsyncCollector.cs

```diff
@@ -33,7 +34,7 @@ public Task AddAsync(SendGridMessage item, CancellationToken cancellationToken =
 
             SendGridHelpers.DefaultMessageProperties(item, _config, _attribute);
 
-            if (item.To == null || item.To.Length == 0)
+            if (item.Personalization?.Count == 0)
```

Comment:
Yeah, your point was valid :-) I've fixed that up and added a test for it. I think if this validation starts getting too crazy or it's a pain to keep up, it's not a big deal to remove it since we can trust the SendGrid server to do the right thing even if we do no validation.